patches/tModLoader/Terraria/Wiring.cs.patch

```diff
@@ -123,7 +123,7 @@
  																for (int num140 = 0; num140 < 200; num140++)
  																{
 -																	if (Main.npc[num140].active && (Main.npc[num140].type == 17 || Main.npc[num140].type == 19 || Main.npc[num140].type == 22 || Main.npc[num140].type == 38 || Main.npc[num140].type == 54 || Main.npc[num140].type == 107 || Main.npc[num140].type == 108 || Main.npc[num140].type == 142 || Main.npc[num140].type == 160 || Main.npc[num140].type == 207 || Main.npc[num140].type == 209 || Main.npc[num140].type == 227 || Main.npc[num140].type == 228 || Main.npc[num140].type == 229 || Main.npc[num140].type == 358 || Main.npc[num140].type == 369 || Main.npc[num140].type == 550))
-+																	if (Main.npc[num140].active && (Main.npc[num140].type == 17 || Main.npc[num140].type == 19 || Main.npc[num140].type == 22 || Main.npc[num140].type == 38 || Main.npc[num140].type == 54 || Main.npc[num140].type == 107 || Main.npc[num140].type == 108 || Main.npc[num140].type == 142 || Main.npc[num140].type == 160 || Main.npc[num140].type == 207 || Main.npc[num140].type == 209 || Main.npc[num140].type == 227 || Main.npc[num140].type == 228 || Main.npc[num140].type == 229 || Main.npc[num140].type == 358 || Main.npc[num140].type == 369 || Main.npc[num140].type == 550 || (Main.npc[num140].townNPC && Main.npc[num140].modNPC?.AddToKingStatue == true)))
++																	if (Main.npc[num140].active && NPCLoader.CanGoToKingStatue(Main.npc[num140]))
  																	{
 -																		array[num139] = num140;
 +																		array.Add(num140);
```

Comment:
Is it safe to do array.Add in these cases? If it does not make any difference I'd rather see it without change to minimize the diff patch

Reply:
I'm doing array.Add here because array isn't an `Array` anymore but a `List`. I changed it because before the limit was of 10 NPCs and modded terraria can get bigger than that, hell, vanilla already has more than 10 that can teleport to the king statue.

Reply:
Alright, good change 👍